Libs/RaptureWebServlet/src/main/java/rapture/server/web/servlet/BaseReflexScriptPageServlet.java

```diff
@@ -171,8 +176,21 @@ void process(String script, Map<String, Object> parameterMap, HttpServletRequest
 
                 ReflexExecutor.runReflexProgram(script, handler, masterParameterMap);
                 resp.setCharacterEncoding("UTF-8");
-                resp.getWriter().append(handler.getOutput());
+                resp.getWriter().append(handler.getOutput()); 
                 resp.setContentType("text/plain");
+                
+                String statusContent = JacksonUtil.jsonFromObject(masterParameterMap.get("status"));
+                logger.debug("Status content is " + statusContent);
+                //ResponseStatus.statusCode is set by default to 200
+                ResponseStatus retStatus = JacksonUtil.objectFromJson(statusContent, ResponseStatus.class);
+                int statusCode = retStatus.getStatusCode();   
+                if (statusCode >= HttpStatus.SC_BAD_REQUEST) {
```

Comment:
Sound probably use HttpServletResponse.SC_BAD_REQUEST here, but it's of course the same. Maybe make it a method in ResponseStatus (i.e. "isError()") so that it's nicely encapsulated there instead of at this level.